File: frontend/src/schemas.ts
export interface User {
  id: number;
  email: string;
  is_active: boolean;
}

export interface Project {
  id: number;
  name: string;
  description?: string;
  owner_id: number;
}

export interface Task {
  id: number;
  title: string;
  description?: string;
  project_id: number;
  assignee_id?: number;
}

File: backend/src/schemas.py
from pydantic import BaseModel

class UserBase(BaseModel):
    email: str

class UserCreate(UserBase):
    password: str

class User(UserBase):
    id: int
    is_active: bool
    projects: list["Project"] = []

    class Config:
        orm_mode = True

class ProjectBase(BaseModel):
    name: str
    description: str | None = None

class ProjectCreate(ProjectBase):
    pass

class Project(ProjectBase):
    id: int
    owner_id: int

    class Config:
        orm_mode = True

class TaskBase(BaseModel):
    title: str
    description: str | None = None

class TaskCreate(TaskBase):
    pass

class Task(TaskBase):
    id: int
    project_id: int
    assignee_id: int | None = None

    class Config:
        orm_mode = True
